BUILD_WORKFLOWS.md

```diff
@@ -0,0 +1,104 @@
+# 🧰 Using the Workflow Builder Image
+
+This container image provides a self-contained environment for building and deploying SonataFlow projects. It wraps a `build.sh` script that simplifies the process from manifest generation to deployment.
+The script can be used directly from Linux/MacOS instead of using the container image, given all of the tools avaiable.
+
+---
+
+## ✅ What the Image Does
+
+When you run the container, it invokes the `build.sh` script, which performs the following:
+
+1. **Generates SonataFlow manifests** using the [`kn-workflow`](https://github.com/kiegroup/kogito-tooling) plugin (v1.35.0 required).
+2. **Builds the workflow image** using either Podman or Docker.
+3. **Optionally deploys** the workflow:
+   - Pushes the built image to the container registry.
+   - Applies the generated manifests to the current Kubernetes cluster using `kubectl`.
+
+---
+
+## 📌 Script Usage
+
+```text
+Usage:
+/app/scripts/build.sh [flags]
+```
+
+### Flags:
+
+| Flag | Description |
+|------|-------------|
+| `-i`, `--image=<string>` | **(Required)** Full image name to build and optionally push (e.g., `quay.io/org/app:tag`). |
+| `-b`, `--builder-image=<string>` | Override the builder image used during the workflow image build. |
+| `-r`, `--runtime-image=<string>` | Override the base image for running the workflow. |
+| `-n`, `--namespace=<string>` | Target Kubernetes namespace. Defaults to current context namespace. |
+| `-m`, `--manifests-directory=<string>` | Directory for outputting the generated manifests. Defaults to `./manifests`. |
+| `-w`, `--workflow-directory=<string>` | Directory containing the workflow's `src/`. Defaults to current directory. |
+| `-P`, `--no-persistence` | Disables persistence configuration in SonataFlow CR. |
+| `--push` | Pushes the workflow image to the specified registry. |
+| `--deploy` | Applies the generated manifests to the cluster. |
+| `-h`, `--help` | Displays usage help. |
+
+---
+
+## 🚀 Running the Container
+
+Here’s a typical command used to build and optionally push the workflow image:
+
+```bash
+podman run --rm \
+  --network=host \
+  --privileged \
+  -v $HOME/.config/containers/auth.json:/root/.config/containers/auth.json:ro \
```

Comment:
I do not have such file locally as I am not using podam
For docker, it appears it's in `~/.docker/config.json`

Also, why not have the build script run the command itslef by introducing a new flag (i.e: `-c | --containerized`)?

Reply:
With docker, it works with
```
docker run --rm     --privileged   -v $HOME/.docker/config.json:/root/.docker/config.json:ro   -v $(pwd)/05_software_template_hello_world/workflow:/workspace   quay.io/orchestrator/orchestrator-workflow-builder:1.35   -i quay.io/gfarache/builder-test:test   -w /workspace   -m /workspace/manifests   --push
```
so no `--network=host otherwise it fails

Reply:
> Podman first searches for the username and password in the ${XDG_RUNTIME_DIR}/containers/auth.json, if they are not valid, Podman then uses any existing credentials found in $HOME/.docker/config.json. If those credentials are not present, Podman creates ${XDG_RUNTIME_DIR}/containers/auth.json 

source: https://docs.podman.io/en/stable/markdown/podman-login.1.html

At it works like that so the `$HOME/.docker/config.json should always be mounted

Reply:
Also, I have the following
```
WARN: Failed to get the git commit hash, skipping tagging with commit hash
```
which makes sense as we are not copying the whole folder but only the workflow directory. Maybe the commit hash should be passed in the build?

Reply:
> Also, why not have the build script run the command itself by introducing a new flag (i.e: -c | --containerized)?

since the script is being invoked from the container. I'd like to avoid adding another script to wrap this one, and using the same script for both purposes (to invoke the build in a container) is missing the purpose IMO.

with this command, you don't need to clone or run download anything.
if you start cloning the repository and such, just install the tools and use the script directly.

If we wanted to drive everything from the script, we were to take a different approach in which all of the tools invocation is done from container, but the podman/docker commands are run directly on the machine and not within another container.

Reply:
ok